### ISSUE TITLE
GeoJsonLayer tesselates polygon data incorrectly 

### ISSUE THREAD

**Issue by Gimir** [visgl/deck.gl#6751]
### Description

I generated geojson data for drawing **day/night layer** on map (link in reproduce section), and when I pass the data to GeoJsonLayer it draws additional polygon or something.
Here is what layer looks like in my case with deck.gl:

<img width="1439" alt="Снимок экрана 2022-03-11 в 16 35 20" src="https://user-images.githubusercontent.com/29613491/157884109-96c580ea-07b7-4469-b299-a8093f7ba7c9.png">

### Flavors

- [ ] React
- [ ] Python/Jupyter notebook
- [ ] MapboxLayer
- [ ] GoogleMapsOverlay
- [ ] CartoLayer
- [ ] DeckLayer/DeckRenderer for ArcGIS

### Expected Behavior

When I test data passed to GeoJsonLayer in geojson.io, it correctly draws given polygon.

<img width="521" alt="Снимок экрана 2022-03-11 в 17 14 19" src="https://user-images.githubusercontent.com/29613491/157884389-eb4f2bea-e602-46be-9638-663ea609238d.png">

### Steps to Reproduce

Here is [link to Sandbox](https://codesandbox.io/s/zealous-pasteur-cbrpb2?file=/src/index.js)
Generating GeoJson data on line - 17
Creating GeoJsonLayer on line - 37

Here is [Gist with full polygon data in Geojson.io](https://gist.github.com/Gimir/c575cbaf784e67f65aa95272d3c853fa)
(to see raw click on <  > button)

Example layer GeoJson data:
```json
{
    "type": "FeatureCollection",
    "features": [
        {
            "type": "Feature",
            "properties": {},
            "geometry": {
                "type": "Polygon",
                "coordinates": [
                    [
                        [
                            -360,
                            90
                        ],
                        [
                            360,
                            90
                        ],
                        ...
                        [
                            -360,
                            90
                        ],
                    ]
               ]
          }
     }
  ]
}
```

### Environment

- Framework version: **8.7.0**
- Browser: **Chrome latest**
- OS: **Mac os 11.3.1**

**Comment by ibgreen** [comment]
I have seen a similar issue, in the FlatgeobufLoader example in loaders.gl (see the spurious triangle in the very south).

<img width="1548" alt="Screen Shot 2022-03-11 at 8 47 51 AM" src="https://user-images.githubusercontent.com/7025232/157911086-c5f6240a-2d50-4502-a2fd-afa3f2f96fde.png">

<img width="1034" alt="Screen Shot 2022-03-11 at 8 49 20 AM" src="https://user-images.githubusercontent.com/7025232/157911252-8290843e-8a20-4617-87a1-0dae3e368130.png">

To repro, go to https://loaders.gl/examples/geopkg and select the flatgeobuf file in the dropdown.

We haven't ruled out that it could be a problem in the flatgeobuf loader but it seems to follow a similar pattern (i.e. geometry close to the pole being incorrectly tessellated).

**Comment by Gimir** [comment]
@ibgreen yes, it seems to be same issue. I tried two layers (PolygonLayer, SolidPolygonLayer), both have this problem. Also I tried different coordinate systems (180 90, 360 90), same thing.

**Comment by kylebarron** [comment]
A couple quick observations:

- My hunch is that the error would be in the polygon triangulation, not the shaders themselves.
- The maximum latitude of web mercator is 85.05113, so rendering data with latitude values outside that band will never be seen. If I use Python's [Shapely](https://github.com/shapely/shapely) in a small script to intersect your geojson data with the box `-180, -85.05113, 180, 85.05113`, then triangulation works fine
- At least for the flatgeobuf example, I wouldn't be surprised if that issue was more with self-intersections in polygons than with handling polar data

**Comment by Gimir** [comment]
@kylebarron it works! Thank you man, I changed latitude to 85 and artifacts are gone.

**Comment by kylebarron** [comment]
Good to know, though I'd still consider this a bug.

**Comment by ibgreen** [comment]
> Good to know, though I'd still consider this a bug.

Yes, or maybe two different bugs. 

We use `earcut` for tesselation and it is advertised as offering very high performance with the down side being that it doesn't guarantee correctness: https://github.com/mapbox/earcut#the-algorithm. 

If one or both of these issues relate to earcut . I suppose one option could be to also offer a slower, more robust tesselation algorithm.

**Comment by felixpalmer** [comment]
In `PolygonTesselator`, we already cut the polygon by Mercator bounds when `wrapLongitude` is set. Would a similar approach work here?

**Comment by kylebarron** [comment]
We could definitely try it. I haven't tried to render polar data in deck.gl myself, so I can't guarantee that will fix this error, but it might.

**Comment by Pessimistress** [comment]
@Gimir If you set `wrapLongitude: true` on the layer, it will be rendered correctly.

Web Mercator projection's latitude limit is around 85.0511 ([source](https://en.wikipedia.org/wiki/Web_Mercator_projection)). 90 is projected to infinity and is usually the culprit of strange tesselation outcomes. In the shader, we clamp latitudes to 89.9 to avoid artifacts, but we don't do it on the CPU (`Viewport.projectFlat`). We should handle 90 gracefully since it is technically valid GeoJSON.

**Comment by Pessimistress** [comment]
Fixed in v8.7.4
